cocos/base/CCScheduler.cpp

```diff
@@ -464,26 +464,15 @@ void Scheduler::schedulePerFrame(const ccSchedulerFunc& callback, void *target,
     HASH_FIND_PTR(_hashForUpdates, &target, hashElement);
     if (hashElement)
     {
-        // check if priority has changed
+        // change priority: should unschedule it first
         if ((*hashElement->list)->priority != priority)
```

Comment:
I think this line should be like
`if (hashElement->entry->priority != priority)`